clang/test/Driver/unsupported-option-gpu.c

```diff
@@ -0,0 +1,7 @@
+/// Some target-specific options are ignored for GPU, so %clang exits with code 0.
+// DEFINE: %{gpu_opts} = --cuda-gpu-arch=sm_60 --cuda-path=%S/Inputs/CUDA/usr/local/cuda --no-cuda-version-check
+// DEFINE: %{check} = %clang -### -c %{gpu_opts} -mcmodel=medium %s
+// RUN: %{check} -fbasic-block-sections=all
```

Comment:
You may want to either rename the file into `.cu` or use `-x cuda`. Otherwise you're running a plain C compilation.

Also, what exactly are we checking here? With `-###` CC1 sub-compilations do not run and, I think, that's where unsupported options were triggering warning/errors. 

You may want to do two actual compilations, one with `--cuda-host-only` and one with `--cuda-device-only -s`.

Reply:
Added `-x cuda`. The test is to show we don't get an error (`err_drv_unsupported_opt_for_target`) when compiling for x86_64 using a device (AMDGPU/NVPTX) when certain target-specified options are specified.

I am not familiar with offloading but specifying `--cuda-host-only` would defeat the purpose.

Reply:
Offloading compilation for these single-source languages pretty much just combines one "host" compilation job with N "Device" compilation jobs. Doing `--offload-device-only` and `--offload-host-only` simply does one part of that. There's probably some flags that behave differently depending on which end you're compiling on, so maybe it would be useful for separating that behavior if needed.